RMS/Reprocess.py

```diff
@@ -465,6 +466,77 @@ def processNight(night_data_dir, config, detection_results=None, nodetect=False)
         log.debug(repr(traceback.format_exception(*sys.exc_info())))
 
 
+    # Generate a timelapse from frames
+    if config.timelapse_generate_from_frames:
+
+        log.info('Generating timelapse from saved frames...')
+        try:
+            frame_dir = os.path.join(config.data_dir, config.frame_dir)
+
+            # Generate timelapse for each day of the year, if not present
+            for year in os.listdir(frame_dir):
+                # Each 'year' is 2024, 2025, ...
+                year_dir = os.path.join(frame_dir, year)
+
+
+                for day in os.listdir(year_dir):
+                    # Each 'day' is 20240923-267, 20240924-268, ...
+                    day_dir = os.path.join(year_dir, day)
+
+                    # Skip if not directory or if inside today's directory
+                    if (not os.path.isdir(day_dir)) or (day == RmsDateTime.utcnow().strftime("%Y%m%d-%j")):
+                        continue
+                    
+                    img_count = 0
+
+
+                    # Checking frames for each hour
+                    for hour in os.listdir(day_dir):
+                        # Each 'hour' is 20240923-267_00, 20240923-267_01, ...
+                        hour_dir = os.path.join(day_dir, hour)
+
+                        # Skip if not directory
+                        if not os.path.isdir(hour_dir):
+                            continue
+
+                        # Count both .jpg and .png files in the hourly subdirectory(s)
+                        img_count += len(glob.glob(os.path.join(hour_dir, '*.jpg')) + \
+                                         glob.glob(os.path.join(hour_dir, '*.png')))
+
+
+                    if img_count < 2:
+                        # Skip this directory if fewer than 2 JPG files are found
+                        continue
+
+                    # Search for current day's timelapse in the corresponding year's directory
+                    found_files = glob.glob(os.path.join(year_dir, f'{day}_frames_timelapse.mp4'))
+
+                    # If not found, generate timelapse for the current day
+                    if not found_files:
+                        log.info(f"No frames timelapse for {day} found in {year_dir}, generating new timelapse...")
+
+                        # Make the name of the timelapse file from day directory
+                        # The day's timelapse is stored in its corresponding year's directory
+                        frames_timelapse_path = os.path.join(year_dir, day + "_frames_timelapse.mp4")
```

Comment:
Should we add the station ID, e.g. XX0001-20241102-307, to the mp4, bz2, and json file names? Otherwise we end up with the name for all stations.